annsa/tests/test_baseclass.py

```diff
@@ -0,0 +1,94 @@
+from annsa.model_classes import BaseClass
+import numpy as np 
+import matplotlib.pyplot as plt 
+from annsa.template_sampling import make_random_spectrum
+
+def construct_baseclass():
+    base = BaseClass()
+
+    return base
+    
+def test_construct_baseclass():
+
+    _ = construct_baseclass()
+    pass
+
+#define the epochs
+epochs = np.arange(1,100,1)
+#define cost function data
+not_learning_cost = [(-1*epoch) + len(epochs) for epoch in epochs] #negative linear
+earlystop_cost = [(len(epochs)/np.sqrt(epoch) + epoch/10 - 4) for epoch in epochs] #1/sqrt(x) + x
+
+
+#create class
+base = construct_baseclass()
+
+#define lamba = 1000
+#define size = 1x1024(the number of channels)
+dim = (1,1024)
+lam = 1000
+random_spectrum = np.random.poisson(lam=lam, size=dim)
+
+#data augmentation unit tests
+def test_default_data_augmentation():
+    data_aug = base.default_data_augmentation(random_spectrum)
+    assert(data_aug.all() == random_spectrum.all()), "Default data augmentation is not identity function."
+
+def test_poisson_data_augmentation():
+    data_aug = base.poisson_data_augmentation(random_spectrum)
+    mean = np.mean(random_spectrum)
+    assert(abs(lam - mean)/lam < 1), "Poisson data augmentation is not poisson sampling."
+
+#check_earlystop unit tests
+def test_check_earlystop_case1():
+    """case 1: earlystopping is turned off"""
+    earlystop_patience = 0
+    epoch = 5
+    stopped = base.check_earlystop(epoch, earlystop_cost[:epoch], earlystop_patience)
+    assert (stopped==False), "Early stopping is turned off, should not have stopped."
```

Comment:
Is there a reason it is atypical to include error messages? Simply because it is slower to come up with effective messages and the traceback will tell you where it broke?